src/main/java/com/eternalcode/parcellockers/gui/implementation/locker/DestinationSelectionGUI.java

```diff
@@ -0,0 +1,135 @@
+package com.eternalcode.parcellockers.gui.implementation.locker;
+
+import com.eternalcode.parcellockers.configuration.implementation.ConfigItem;
+import com.eternalcode.parcellockers.configuration.implementation.PluginConfiguration;
+import com.eternalcode.parcellockers.gui.GuiView;
+import com.eternalcode.parcellockers.gui.PaginatedGuiRefresher;
+import com.eternalcode.parcellockers.locker.Locker;
+import com.eternalcode.parcellockers.locker.repository.LockerPageResult;
+import com.eternalcode.parcellockers.locker.repository.LockerRepository;
+import com.eternalcode.parcellockers.shared.ExceptionHandler;
+import com.eternalcode.parcellockers.shared.Page;
+import dev.triumphteam.gui.guis.Gui;
+import dev.triumphteam.gui.guis.GuiItem;
+import dev.triumphteam.gui.guis.PaginatedGui;
+import net.kyori.adventure.text.minimessage.MiniMessage;
+import org.bukkit.entity.Player;
+import org.bukkit.plugin.Plugin;
+import org.bukkit.scheduler.BukkitScheduler;
+
+import java.util.List;
+import java.util.UUID;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Supplier;
+
+public class DestinationSelectionGUI extends GuiView {
+
+    private static final int WIDTH = 7;
+    private static final int HEIGHT = 4;
+    private static final Page FIRST_PAGE = new Page(0, WIDTH * HEIGHT);
+
+
+    private final Plugin plugin;
+    private final BukkitScheduler scheduler;
+    private final PluginConfiguration config;
+    private final MiniMessage miniMessage;
+    private final LockerRepository lockerRepository;
+    private final ParcelSendingGUI sendingGUI;
+    private final ParcelSendingGUIState state;
+
+    public DestinationSelectionGUI(Plugin plugin, BukkitScheduler scheduler, PluginConfiguration config, MiniMessage miniMessage, LockerRepository lockerRepository, ParcelSendingGUI sendingGUI, ParcelSendingGUIState state) {
+        this.plugin = plugin;
+        this.scheduler = scheduler;
+        this.config = config;
+        this.miniMessage = miniMessage;
+        this.lockerRepository = lockerRepository;
+        this.sendingGUI = sendingGUI;
+        this.state = state;
+    }
```

Comment:
It is probably possible to divide it so that you have less dependence here.

Reply:
I already shrinked this by 50% by isolating dependencies into the `ParcelSendingGUIState`.